Source/Particles/ParticleCreation/SmartUtils.cpp

```diff
@@ -44,12 +50,21 @@ SmartCopyTag getSmartCopyTag (const NameMap& src, const NameMap& dst) noexcept
         {
             // name is in both...
             tag.common_names.push_back(i_src->first);
-            tag.src_comps.push_back(i_src->second);
-            tag.dst_comps.push_back(i_dst->second);
+            h_src_comps.push_back(i_src->second);
+            h_dst_comps.push_back(i_dst->second);
             ++i_src;
             ++i_dst;
         }
     }
 
+    tag.src_comps.resize(h_src_comps.size());
+    amrex::Gpu::htod_memcpy_async(tag.src_comps.data(), h_src_comps.data(),
+                                  h_src_comps.size()*sizeof(int));
+
+    tag.dst_comps.resize(h_dst_comps.size());
+    amrex::Gpu::htod_memcpy_async(tag.dst_comps.data(), h_dst_comps.data(),
+                                  h_dst_comps.size()*sizeof(int));
```

Comment:
Maybe that is simpler:

```suggestion
    amrex::Gpu::copyAsync(amrex::Gpu::hostToDevice, h_src_comps.begin(), h_src_comps.end(), tag.src_comps.begin());

    tag.dst_comps.resize(h_dst_comps.size());
    amrex::Gpu::copyAsync(amrex::Gpu::hostToDevice, h_dst_comps.begin(), h_dst_comps.end(), tag.dst_comps.begin());
```